datahub-web-react/src/app/lineage/LineageEntityNode.tsx

```diff
@@ -81,13 +82,20 @@ export default function LineageEntityNode({
     onEntityCenter: (EntitySelectParams) => void;
     onHover: (EntitySelectParams) => void;
     onDrag: (params: EntitySelectParams, event: React.MouseEvent) => void;
-    onExpandClick: (LineageExpandParams) => void;
+    onExpandClick: (data: EntityAndType) => void;
     direction: Direction;
     nodesToRenderByUrn: Record<string, VizNode>;
 }) {
     const { expandTitles } = useContext(LineageExplorerContext);
     const [isExpanding, setIsExpanding] = useState(false);
     const [expandHover, setExpandHover] = useState(false);
+    const { getAsyncEntity, asyncData } = useLazyGetEntityQuery();
+
+    useEffect(() => {
+        if (asyncData) {
+            onExpandClick(asyncData);
+        }
+    }, [asyncData, onExpandClick]);
```

Comment:
nice! this is a surprisingly small change

Reply:
yeah I was pleasantly surprised!